test_util/helpers.py

```diff
@@ -48,85 +47,56 @@ def authenticate(self, cluster):
+        if self.ca_cert_path and 'verify' not in kwargs:
             kwargs['verify'] = self.ca_cert_path
 
+        if self.api_base:
+            path = path_join(self.api_base, path)
+
+        request_url = path_join(host_url if host_url else self.default_host_url, path)
         headers.update(kwargs.pop('headers', {}))
-        url = self.get_url(node=node, path=path, port=port)
-        logging.info('Request method {}: {}'.format(method, url))
+        logging.info('Request method {}: {}'.format(request_fn.__name__, request_url))
         logging.debug('Reqeust kwargs: {}'.format(kwargs))
         logging.debug('Request headers: {}'.format(headers))
-        return requests.request(method, url, headers=headers, **kwargs)
+        return request_fn(request_url, headers=headers, **kwargs)
 
-    def get_url(self, node, path, port=None):
-        """
-        Args:
-            path: (str) URL path to request if self.api_base is set it will be prepended
-            node: (str) the hostname of the node to be requested from, if node=None,
-                then public cluster address (see environment DCOS_DNS_ADDRESS)
-            port: (int) port to be requested at. If port=None, the default port
-                for that given node type will be used
-        Returns:
-            fully-qualified URL string for this API
+    def wrapped_request(self, request_fn, *args, **kwargs):
+        """Thin wrapper to allow wrapping requests dynamically by mapping
+        a function to self.request_wrapper
         """
-        if node is None:
-            node = self.cluster.dns_host
-            role = 'master'
-        else:
-            if node in self.cluster.masters:
-                role = 'master'
-            elif node in self.cluster.all_slaves:
-                role = 'agent'
-            else:
-                raise Exception('Node {} is not recognized within the DC/OS cluster'.format(node))
-
-        if self.api_base:
-            path = '{}/{}'.format(self.api_base.rstrip('/'), path.lstrip('/'))
-
-        if port is None:
-            port = ADMINROUTER_PORT_MAPPING[role][self.cluster.scheme]
-
-        if (port == 80 and self.cluster.scheme == 'http') or (port == 443 and self.cluster.scheme == 'https'):
-            netloc = node
-        else:
-            netloc = '{}:{}'.format(node, port)
-
-        return '{}://{}/{}'.format(self.cluster.scheme, netloc, path.lstrip('/'))
-
-    get = functools.partialmethod(api_request, 'get')
-    post = functools.partialmethod(api_request, 'post')
-    put = functools.partialmethod(api_request, 'put')
-    delete = functools.partialmethod(api_request, 'delete')
-    options = functools.partialmethod(api_request, 'options')
-    head = functools.partialmethod(api_request, 'head')
-    patch = functools.partialmethod(api_request, 'patch')
-    delete = functools.partialmethod(api_request, 'delete')
+        return self.request_wrapper(request_fn, *args, **kwargs)
```

Comment:
I reviewed this piece too. The change is easy to understand. So, this is fine with me.